src/ZfcUser/Authentication/Adapter/AdapterChain.php

```diff
@@ -2,139 +2,85 @@
 
 namespace ZfcUser\Authentication\Adapter;
 
-use Zend\Authentication\Adapter\AdapterInterface;
-use Zend\Authentication\Result as AuthenticationResult;
-use Zend\EventManager\Event;
-use Zend\Stdlib\RequestInterface as Request;
-use Zend\Stdlib\ResponseInterface as Response;
-use ZfcBase\EventManager\EventProvider;
-use ZfcUser\Exception;
-
-class AdapterChain extends EventProvider implements AdapterInterface
+use Zend\Authentication\Adapter\AbstractAdapter;
+use Zend\Stdlib\PriorityList;
+use Zend\Authentication\Result;
+use Zend\EventManager\EventManagerAwareTrait;
+
+/**
+ * Chainable authentication adapter for Zend\Authentication
+ * 
+ * Allows multiple authentication adapters to be tried in succession, 
+ * breaking the chain on the first adapter to return a successful result
+ */
+class AdapterChain extends AbstractAdapter
 {
+    use EventManagerAwareTrait;
+    
     /**
-     * @var AdapterChainEvent
-     */
-    protected $event;
-
-    /**
-     * Returns the authentication result
-     *
-     * @return AuthenticationResult
+     * @var PriorityList
      */
-    public function authenticate()
+    protected $adapters;
+        
+    public function __construct()
     {
-        $e = $this->getEvent();
-
-        $result = new AuthenticationResult(
-            $e->getCode(),
-            $e->getIdentity(),
-            $e->getMessages()
-        );
-
-        $this->resetAdapters();
-
-        return $result;
+        $this->adapters = new PriorityList();
+        $this->adapters->isLIFO(false);
```

Comment:
Stdlib is a bit of a mess in that regard.  SplPriorityQueue and SplQueue are wrappers around the SPL classes to make them serializable.  PriorityQueue uses `Zend\Stdlib\SplPriorityQueue` internally to get around SplPriorityQueue's destructive iterator, while PriorityList is an entirely standalone implementation.  PriorityList with LIFO=false is functionally equivalent to PriorityQueue, just implemented differently.  PriorityList with LIFO=true is technically a priority stack, but there is no PriorityStack in Stdlib.  The only difference between PriorityStack and PriorityQueue is how they break priority ties.  As far as I can tell it's all necessitated by [odd behaviours in the Spl classes](https://mwop.net/blog/253-Taming-SplPriorityQueue.html).  And all of this has absolutely nothing to do with this PR, I just got off-track but I spent the time writing it so I might as well post it.

I'm going to try to sub in `\SplPriorityQueue` to see if it will work for this use case.  If not, I'll sub in `Zend\Stdlib\PriorityQueue` which is equivalent to what I have currently but will cause no future head-scratching over `isLIFO(false)`.